### PR TITLE
Make inspec a variable so version can be set

### PR DESCRIPTION
**Pull request by TheLunaticScripter**
Signed-off-by: John Snow <thelunaticscripter@outlook.com>

Currently InSpec on the windows and linux scaffolding is hard coded to `chef/inspec` or `stuartpreston/inspec` this needs to get changed to a variable that will allow users to pin to a version of inspec or use an alternate package.

### CODE REVIEW

**Comment by TheLunaticScripter** [comment]
#171 this will allow a workaround for this issue so that users can pin to `chef/inspec/4.16.0` while the issue is being fixed in core inspec.

File: scaffolding-chef-inspec/tests/user-linux-api/tests/test.bats [hunk 1]
```diff
@@ -0,0 +1,22 @@
+SCAFFOLD_PKG_INSPEC_CLIENT_VERSION="$(echo "${scaffold_inspec_client}" | cut -d/ -f3)"
+
+@test "API: scaffold_cacerts matches run hook core/cacerts" {
+  result="$(grep '^export SSL_CERT_FILE.*' /hab/svc/${TEST_PKG_NAME}/hooks/run | cut -d/ -f4-5)"
+  [ "${result}" = "core/cacerts" ]
+}
+
+@test "API: scaffold_cacerts matches run hook core/cacerts" {
```
Comment (lancewf) [inline scaffolding-chef-inspec/tests/user-linux-api/tests/test.bats]: This is a duplicate test from the one above.

File: scaffolding-chef-inspec/lib/scaffolding.ps1 [hunk 2]
```diff
@@ -2,12 +2,16 @@
 # A scaffolding for an InSpec profile
 #
 
+if(!$scaffold_inspec_client){
+    $scaffold_inspec_client = "stuartpreston/inspec"
```
Comment (lancewf) [inline scaffolding-chef-inspec/lib/scaffolding.ps1]: Should we default to "chef/inspec"?

Comment (TheLunaticScripter) [reply scaffolding-chef-inspec/lib/scaffolding.ps1]: Unfortunately, we don't have a windows `chef/inspec` package yet this is being worked on at the moment.

File: scaffolding-chef-inspec/tests/user-linux-api/habitat/plan.sh [hunk 3]
```diff
@@ -0,0 +1,5 @@
+pkg_name=user-linux-default
+pkg_version=0.1.0
+pkg_origin=chef
+pkg_scaffolding="ci/scaffolding-chef-inspec"
```
Comment (lancewf) [inline scaffolding-chef-inspec/tests/user-linux-api/habitat/plan.sh]: I don't understand how the "ci" origin works for the `pkg_scaffolding="ci/scaffolding-chef-inspec"`

Comment (lancewf) [reply scaffolding-chef-inspec/tests/user-linux-api/habitat/plan.sh]: I kind of see what is happening here. The test builds "scaffolding-chef-inspec" with the "ci" origin and then the test is run with that build.

Comment (TheLunaticScripter) [reply scaffolding-chef-inspec/tests/user-linux-api/habitat/plan.sh]: Yep this is cause we need to verify the scaffolding using a user plan but I don't think we want to build something that would go into unstable on every PR.

**Comment by TheLunaticScripter** [comment]
It can't that is a thing I would like to get working. I think it has something to do with the supervisor `CTL_SECRET` but I haven't been able to figure it out.

**Comment by robbkidd** [comment]
@lancewf 
> hab-studio/0.77.0
 👴 😢

**Pull request merged**